AsyncDisplayKit/ASTextNode.mm

```diff
@@ -790,6 +822,8 @@ - (UIImage *)placeholderImage
     return nil;
   }
   
+  std::lock_guard<std::recursive_mutex> l(_textLock);
```

Comment:
Yeah, holding this lock at this scope while we call below into other methods that also take the same lock makes me nervous.